update-version.sh

```diff
@@ -1,65 +1,99 @@
+#
+# update-version.sh - Updates or checks README.md version
+#
+# This script is used for two main purposes:
+# 1. Update the version in a module's README.md file (normal mode)
+# 2. Check if the version in README.md matches a specified version (--check mode)
+#
+# It's primarily used by the GitHub Actions workflow that runs when tags are pushed.
 
 set -euo pipefail
 
-current_tag=$(git describe --tags --abbrev=0)
-
-# Increment the patch version
-LATEST_TAG=$(echo "$current_tag" | sed 's/^v//' | awk -F. '{print $1"."$2"."$3+1}') || exit $?
+# Check if we're in check-only mode
+CHECK_ONLY=false
+if [[ "$1" == "--check" ]]; then
+  CHECK_ONLY=true
+  shift
+fi
 
-# List directories with changes that are not README.md or test files
-mapfile -t changed_dirs < <(git diff --name-only "$current_tag" -- ':!**/README.md' ':!**/*.test.ts' | xargs dirname | grep -v '^\.' | sort -u)
+# Validate we have the right number of arguments
+if [[ "$#" -ne 2 ]]; then
+  echo "Usage: ./update-version.sh [--check] module-name X.Y.Z"
+  exit 1
+fi
 
-echo "Directories with changes: ${changed_dirs[*]}"
+MODULE_NAME="$1"
+VERSION="$2"
 
-# Iterate over directories and update version in README.md
-for dir in "${changed_dirs[@]}"; do
-  if [[ -f "$dir/README.md" ]]; then
-    file="$dir/README.md"
-    tmpfile=$(mktemp /tmp/tempfile.XXXXXX)
-    awk -v tag="$LATEST_TAG" '
-      BEGIN { in_code_block = 0; in_nested_block = 0 }
-      {
-        # Detect the start and end of Markdown code blocks.
-        if ($0 ~ /^```/) {
-          in_code_block = !in_code_block
-          # Reset nested block tracking when exiting a code block.
-          if (!in_code_block) {
-            in_nested_block = 0
-          }
-        }
+# Validate version format (X.Y.Z)
+if ! [[ "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+  echo "Error: Version must be in format X.Y.Z"
+  exit 1
+fi
 
-        # Handle nested blocks within a code block.
-        if (in_code_block) {
-          # Detect the start of a nested block (skipping "module" blocks).
-          if ($0 ~ /{/ && !($1 == "module" || $1 ~ /^[a-zA-Z0-9_]+$/)) {
-            in_nested_block++
-          }
+# Check if module directory exists
+if [[ ! -d "$MODULE_NAME" || ! -f "$MODULE_NAME/README.md" ]]; then
+  echo "Error: Module directory not found or missing README.md"
+  exit 1
+fi
 
-          # Detect the end of a nested block.
-          if ($0 ~ /}/ && in_nested_block > 0) {
-            in_nested_block--
-          }
+# Extract version from README.md file
+extract_version() {
+  # This finds version lines like: version = "1.2.3"
+  grep -o 'version *= *"[0-9]\+\.[0-9]\+\.[0-9]\+"' "$1" | head -1 | grep -o '[0-9]\+\.[0-9]\+\.[0-9]\+' || echo "0.0.0"
+}
 
-          # Update "version" only if not in a nested block.
-          if (!in_nested_block && $1 == "version" && $2 == "=") {
-            sub(/"[^"]*"/, "\"" tag "\"")
-          }
+# Update version in README.md file
+update_version() {
+  local tmpfile=$(mktemp)
```

Comment:
```suggestion
  local file=$1 latest_tag=$2 tmpfile
  tmpfile=$(mktemp)
```
Let's keep it readable and `local` masks non-zero exit codes.